docs/azure/sdk/snippets/authentication/best-practices/CCA/Program.cs

```diff
@@ -2,24 +2,35 @@
 using Azure.Identity;
 using Microsoft.Extensions.Azure;
 
-var clientId = "<user-assigned-client-id>";
-var builder = WebApplication.CreateBuilder(args);
+string clientId = "<user-assigned-client-id>";
+string storageAccountName = "<account-name>";
+string keyVaultName = "<vault-name>";
+WebApplicationBuilder builder = WebApplication.CreateBuilder(args);
 
 #region snippet_credential_reuse_AspNetCore
 builder.Services.AddAzureClients(clientBuilder =>
 {
-    clientBuilder.AddSecretClient(new Uri("<key-vault-url>"));
-    clientBuilder.AddBlobServiceClient(new Uri("<blob-storage-url>"));
+    clientBuilder.AddSecretClient(
+        new Uri($"https://{keyVaultName}.vault.azure.net"));
+    clientBuilder.AddBlobServiceClient(
+        new Uri($"https://{storageAccountName}.blob.core.windows.net"));
 
     string? clientId = builder.Configuration["UserAssignedClientId"];
+    TokenCredential credential;
 
-    TokenCredential credential = clientId is not null
-        ? new ManagedIdentityCredential(
-            ManagedIdentityId.FromUserAssignedClientId(clientId))
-        : new ChainedTokenCredential(
+    if (builder.Environment.IsProduction() && clientId is not null)
```

Comment:
I think we'd rather have this fail in prod than to fall through to the dev credentials. 

```suggestion
    if (builder.Environment.IsProduction())
```